toolchain/sem_ir/function.cpp

```diff
@@ -36,39 +63,49 @@ auto GetCalleeFunction(const File& sem_ir, InstId callee_id,
 
   if (auto specific_function =
           sem_ir.insts().TryGetAs<SpecificFunction>(callee_id)) {
-    result.resolved_specific_id = specific_function->specific_id;
+    fn.resolved_specific_id = specific_function->specific_id;
     callee_id = specific_function->callee_id;
   }
 
   // Identify the function we're calling by its type.
   auto val_id = sem_ir.constant_values().GetConstantInstId(callee_id);
   if (!val_id.has_value()) {
-    return result;
+    return {.info = CalleeFunction::Other()};
   }
   auto fn_type_inst =
       sem_ir.types().GetAsInst(sem_ir.insts().Get(val_id).type_id());
 
   if (auto cpp_overload_set_type = fn_type_inst.TryAs<CppOverloadSetType>()) {
-    result.cpp_overload_set_id = cpp_overload_set_type->overload_set_id;
-    return result;
+    return {.info = CalleeFunction::CppOverloadSet{
```

Comment:
Should we CHECK that we don't have a `resolved_specific_id` here, rather than discarding it if we do?